test/Integration/MiddlewareDispatchTest.php

```diff
@@ -78,11 +83,14 @@ public function testDispatchesMiddleware(): void
     public function testMiddlewareDispatchTriggersSharedEventOnMiddlewareController(): void
     {
         $sharedEm = $this->application->getEventManager()->getSharedManager();
+        assert($sharedEm !== null);
         $services = $this->application->getServiceManager();
-        $services->get('Request')->setUri('http://example.local/middleware');
+        /** @var Request $request */
+        $request = $services->get('Request');
+        $request->setUri('http://example.local/middleware');
         $services->setService('MiddlewareMock', new Middleware());
 
-        /** @var callable&MockObject $listener */
+        /** @var Closure&MockObject $listener */
```

Comment:
This does not seem right. This is not a Closure.

Reply:
callable object with `__invoke()` should be fine. Psalm bug?

Reply:
> 
> 
> callable object with `__invoke()` should be fine. Psalm bug?

Yep. I struggled forever with this one. I either get a `Intersection types must all be objects` error or `callable&MockObject is not a valid type`.
I seems that `callable` can not be used for intersection types. If I use the pipe `|` it complains too (`expects callable, PHPUnit\Framework\MockObject\MockObject&stdClass provided`)

Can I just suppress it? 🙂

Reply:
Can you check psalm issues and report it if not there yet?

Reply:
The way to workaround it "properly" here might be to document listener as callable after the mock object calls were performed

Reply:
> The way to workaround it "properly" here might be to document listener as callable after the mock object calls were performed

That works, but then `phpcs` complains `No assignment to $listener variable before or after the documentation comment.`